0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -0,0 +1,82 @@
+# Pre-Shared Key
+
+The Pre-Shared Key (PSK) protocol is an end-to-end transport protocol, used by the sender and receiver of an ILP payment to decide on a condition and fulfillment for a payment.
+
+As the name suggests, PSK relies on a pre-shared secret key that is shared beforehand. An application-layer protocol like SPSP could pass this pre-shared secret key over HTTPS. Within one node process, the pre-shared key could be passed via a simple variable.
+
+An advantage of PSK is that the pre-shared secret key only needs to be shared once. So long as both parties possess this key and are listening for transfers, they can send payments between one another.
+
+A disadvantage of PSK is that it is repudiable. Knowledge of a payment's fulfillment cannot be used as cryptographic proof that the receiver completed the payment, because both sender and receiver know how to generate the fulfillment with their pre-shared secret key.
+
+## Flow
+
+0. The pre-shared secret key is shared out of band.
+0. The sender creates an inner (PRIVATE) PSK request, with information on the expiry and any additional data enclosed within.
+0. The sender creates a random Key-Id value, and calculates an HMAC of the Key-Id using the pre-shared secret key as the key. This HMAC will be referred to as the payment key.
```

Comment:
I think we should leave the `receiverId` in the ILP address, because it will be used by different receivers to figure out if a payment is for them. However, we should take the uuid out of the address and move that into a transport-layer header. I think that should probably be in the `Key-Id` field (or a header with a different name that replaces that)